lib/matplotlib/backends/qt_compat.py

```diff
@@ -49,6 +49,10 @@
     # user has imported PySide before importing mpl
     QT_API = QT_API_PYSIDE
 
+if 'PySide2' in sys.modules:
+    # user has imported PySide before importing mpl
+    QT_API = QT_API_PYSIDE2
+
```

Comment:
These things are going to be mutually exclusive due to restrictions from the libraries.

This could be cleaned up, but I think is out of scope of _this_ PR.